server/src/main/java/org/opensearch/index/remote/RemoteStoreUtils.java

```diff
@@ -69,4 +76,29 @@ public static String getSegmentName(String filename) {
 
         return filename.substring(0, endIdx);
     }
+
+    public static Set<FileSnapshot.TransferFileSnapshot> getUploadBlobsFromSnapshot(
+        TransferSnapshot transferSnapshot,
+        FileTransferTracker fileTransferTracker
+    ) {
+        Set<FileSnapshot.TransferFileSnapshot> toUpload = new HashSet<>(transferSnapshot.getTranslogTransferMetadata().getCount());
+        toUpload.addAll(fileTransferTracker.exclusionFilter(transferSnapshot.getTranslogFileSnapshots()));
+        toUpload.addAll(fileTransferTracker.exclusionFilter((transferSnapshot.getCheckpointFileSnapshots())));
+
+        return toUpload;
+    }
+
+    public static long getCurrentSystemNanoTime() {
```

Comment:
Why cannot we access the System.nanoTime() directly wherever we need? This is already accessed in a manner that it can be accessed directly everywhere.

Reply:
Addressed. Resolving this thread. Please reopen if any changes are needed.